### PR TITLE
Add even more GHA log groups

### PR DESCRIPTION
**Pull request by jyn514**
This also adds a dynamic check that we don't emit nested groups, since GHA currently doesn't support them.

r? @oli-obk

### CODE REVIEW

File: src/tools/build_helper/src/ci.rs [hunk 1]
```diff
@@ -45,6 +49,11 @@ pub mod gha {
         } else {
             eprintln!("{name}")
         }
+        // https://github.com/actions/toolkit/issues/1001
+        assert!(
+            !GROUP_ACTIVE.swap(true, Ordering::Relaxed),
+            "nested groups are not supported by GHA!"
+        );
```
Comment (oli-obk) [inline src/tools/build_helper/src/ci.rs:56]: I should probably do this in ui_test, too

**Review by oli-obk — APPROVED** [review]

**Comment by oli-obk** [comment]
@bors r+ rollup

wow, logs look so accessible now

**Comment by matthiaskrgr** [comment]
@bors r-
https://github.com/rust-lang/rust/pull/113554#issuecomment-1629765366

**Comment by jyn514** [comment]
try build got merged and this was tested before the tree was closed :disappointed: https://github.com/rust-lang/rust/commit/bef6ff618f17398775e9d8cd23a6f47d983869dc https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/Cannot.20get.20rust-dev-nightly.20from.20ci-artifacts.2Erust-lang.2Eorg

**Comment by jyn514** [comment]
this doesn't affect how the compiler was built at all, the perf regression must be spurious.

**Comment by pnkfelix** [comment]
> this doesn't affect how the compiler was built at all, the perf regression must be spurious.

I agree. Marking as triaged.

@rustbot label: +perf-regression-triaged

**Pull request merged**